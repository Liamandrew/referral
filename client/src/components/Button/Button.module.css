.button {
    text-transform: uppercase;
    height: 68px;
    width: 100%;
    font-size: 18px;
    box-shadow: 4px 4px 4px #cdcdcd;
    border-radius: 4px;
}

.primaryButton {
    composes: button;
    background-color: #00e576;
    color: white;
    font-weight: 700;
    border: none;
}

.secondaryButton {
    composes: button;
    background-color: white;
    color: #868383;
    font-weight: 900;
    border: 1px solid #cdcdcd;
}
