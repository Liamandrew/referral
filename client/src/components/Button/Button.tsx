import React from "react";
import style from "./Button.module.css";

interface ButtonProps {
    variant: "primary" | "secondary";
    type?: "button" | "submit";
    label: string;
    onClick?: () => void;
}

const Button: React.FC<ButtonProps> = ({
    variant,
    label,
    onClick,
    type = "button",
}) => (
    <button type={type} className={style[`${variant}Button`]} onClick={onClick}>
        {label}
    </button>
);

export { Button };
