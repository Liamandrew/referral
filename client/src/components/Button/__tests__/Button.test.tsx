import React from "react";
import { render } from "@testing-library/react";
import { Button } from "../Button";

describe("<Button>", () => {
    test("variant primary should match snapshot", () => {
        const { container } = render(
            <Button variant="primary" label="label" />
        );

        expect(container).toMatchSnapshot();
    });

    test("variant secondary should match snapshot", () => {
        const { container } = render(
            <Button variant="secondary" label="label" />
        );

        expect(container).toMatchSnapshot();
    });
});
