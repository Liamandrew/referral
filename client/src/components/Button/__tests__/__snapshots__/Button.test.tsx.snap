// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> variant primary should match snapshot 1`] = `
<div>
  <button
    class="primaryButton"
    type="button"
  >
    label
  </button>
</div>
`;

exports[`<Button> variant secondary should match snapshot 1`] = `
<div>
  <button
    class="secondaryButton"
    type="button"
  >
    label
  </button>
</div>
`;
